### ISSUE TITLE
unavailability

### ISSUE THREAD

**Issue by torkus** [vargen2/Addon#54]
Hey, @vargen2 - any chance we can get a download link happening again? Cheers

**Comment by vargen2** [comment]
Program does not work atm. Have to find a workaround for cloudflare ddos protection.

**Comment by torkus** [comment]
Are you scraping the HTML or using the API? 

I've never had a problem with their cloudflare protection before, but I have heard of others having a problem.

An unofficial interface for their API can be found here: https://twitchappapi.docs.apiary.io/#/reference

These aren't great, but here is an [example of searching](https://github.com/ogri-la/wowman/blob/develop/src/wowman/curseforge_api.clj#L142-L145) and an [example of fetching an addon's details](https://github.com/ogri-la/wowman/blob/develop/src/wowman/curseforge_api.clj#L104-L105)

Layday+Instawow (Python) use it too: https://github.com/layday/instawow/blob/master/instawow/resolvers.py#L96

**Comment by vargen2** [comment]
@torkus Yes I'm scraping. Is that API official and free to use. Just the documentation that's unofficial?

**Comment by layday** [comment]
The API is used internally by the Twitch client - it's not meant for public use hence the lack of official documentation.  I can't imagine they'd rather have you scrape the site than tap into their API though if that's what you're curious about.

**Comment by torkus** [comment]
>  Is that API official and free to use. Just the documentation that's unofficial?

correct on all counts. 

> it's not meant for public use 

also definitely the case, but by being publicly available and also undocumented it means they can avoid any sort of user support or guarantees of stability. The results are pretty messy tbh and I think in it's current state they've merged several other APIs into it, but json is much more convenient to parse.

I was making about 330+ requests to the website before to generate the catalog wowman uses. With the API it's down to about 28.
